lib/ansible/playbook/collectionsearch.py

```diff
@@ -32,9 +36,18 @@ def _ensure_default_collection(collection_list=None):
 class CollectionSearch:
 
     # this needs to be populated before we can resolve tasks/roles/etc
-    _collections = FieldAttribute(isa='list', listof=string_types, priority=100, default=_ensure_default_collection)
+    _collections = FieldAttribute(isa='list', listof=string_types, priority=100, default=_ensure_default_collection,
+                                  always_post_validate=True, static=True)
 
     def _load_collections(self, attr, ds):
+        # This duplicates static attr checking logic from post_validate()
+        # because if the user attempts to template a collection name, it will
+        # error before it ever gets to the post_validate() warning.
+        for collection_name in ds:
+            if is_template(collection_name, Environment()):
```

Comment:
Instantiate `Environment` 1 time and re-use it.

To address other potential questions here, we don't allow for things like `variable_start` to be overriden in the playbook "DSL", as such, we don't need to account for that here.  Although we could in the future, but I'm not concerned.

Also, `Environment.lex` doesn't care about undefined vars, or non-existent filters, etc...So we don't need that context anyway.

Reply:
I think this code may actually need to be moved after `_ensure_default_collection`.  `_ensure_default_collection` will make sure the value is formatted correctly.  In the case where someone just does `collections: whatever` it will be turned into a list as well.  So maybe it needs moved all the way to the bottom.